src/Microsoft.DotNet.Web.ProjectTemplates/content/RazorPagesWeb-CSharp/.template.config/vs-2017.3.host.json

```diff
@@ -25,24 +25,115 @@
       "authenticationType": "NoAuth"
     },
     {
-      "auth": "Individual",
+      "auth": "None",
+      "interalAuth": "IndividualB2C",
       "authenticationType": "IndividualAuth",
-      "b2cAuthenticationOptions": "CloudExisting"
+      "b2cAuthenticationOptions": "CloudExisting",
+      "snippets": [
+        {
+          "templateId": "Microsoft.Web.Authentication.Internal.CSharp.2.1",
+          "outputs": [
+            {
+              "name": "__ActiveDirectorySettings__",
+              "destinationSymbol": "__ActiveDirectorySettings__"
+            },
+            {
+              "name": "__ActiveDirectoryStartup__",
+              "destinationSymbol": "__ActiveDirectoryStartup__"
+            },
+            {
+              "name": "internalAuth",
+              "destinationSymbol": "internalAuth"
+            }
+
+          ],
+          "parameters": {
+            "Domain": "Domain",
+            "ClientId": "ClientId",
+            "CallbackPath": "CallbackPath",
+            "SignUpSignInPolicyId": "SignupSignInPolicyId",
+            "ResetPasswordPolicyId": "ResetPasswordPolicyId",
+            "EditProfilePolicyId": "EditProfilePolicyId"
+          },
+          "constants": {
+            "auth": "IndividualB2C",
+            "WebAuth": "true"
+          }
+        }
+      ]
     },
     {
       "auth": "Individual",
       "authenticationType": "IndividualAuth",
       "b2cAuthenticationOptions": "Local"
     },
     {
-      "auth": "SingleOrg",
+      "auth": "None",
       "authenticationType": "OrgAuth",
-      "orgAuthenticationOptions": "SSO"
+      "orgAuthenticationOptions": "SSO",
+      "snippets": [
+        {
+          "templateId": "Microsoft.Web.Authentication.Internal.CSharp.2.1",
+          "outputs": [
+            {
+              "name": "__ActiveDirectorySettings__",
+              "destinationSymbol": "__ActiveDirectorySettings__"
+            },
+            {
+              "name": "__ActiveDirectoryStartup__",
+              "destinationSymbol": "__ActiveDirectoryStartup__"
+            },
+            {
+              "name": "internalAuth",
+              "destinationSymbol": "internalAuth"
+            }
+
+          ],
+          "parameters": {
+            "Domain": "Domain",
+            "ClientId": "ClientId",
+            "TenantId": "TenantId",
+            "CallbackPath": "CallbackPath"
```

Comment:
Could you avoid passing CallbackPath as a parameter for OrgAuth SSO and MMO only? it appears that we were not passing it explicitly before, instead the templates were just using their default value specified in the template.json. I'll get this fixed for preview 4